spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/log/Slf4JSpanLoggerTest.java

```diff
@@ -58,29 +59,119 @@ public void setup() {
 	public void should_set_entries_to_mdc_from_span() throws Exception {
 		Scope scope = this.slf4jCurrentTraceContext.newScope(this.span.context());
 
-		assertThat(MDC.get("X-B3-TraceId")).isEqualTo(span.context().traceIdString());
-		assertThat(MDC.get("traceId")).isEqualTo(span.context().traceIdString());
+		assertMDCInfoEqualToSpanInfo(span.context());
 
 		scope.close();
 
-		assertThat(MDC.get("X-B3-TraceId")).isNullOrEmpty();
-		assertThat(MDC.get("traceId")).isNullOrEmpty();
+		assertMDCInfoNullOrEmpty();
+	}
+
+	@Test
+	public void should_set_entries_to_mdc_from_two_spans() throws Exception {
+		Scope scope = this.slf4jCurrentTraceContext.newScope(this.span.context());
+
+		assertMDCInfoEqualToSpanInfo(span.context());
+
+		Scope scopeInner = this.slf4jCurrentTraceContext.newScope(this.span.context());
+
+		assertMDCInfoEqualToSpanInfo(span.context());
+
+		scopeInner.close();
+
+		assertMDCInfoEqualToSpanInfo(span.context());
+
+		scope.close();
+
+		assertMDCInfoNullOrEmpty();
+	}
+
+	@Test
+	public void should_set_entries_to_mdc_from_two_spans1() throws Exception {
+		Scope scope = this.slf4jCurrentTraceContext.newScope(this.span.context());
```

Comment:
just a small minor suggestion:
`Scope` implements `Closeable` and can be used with `try-with-resources` style

Reply:
done, and the test failure seems not my problem?